batch/snippets/src/main/java/com/example/batch/CreateBatchCustomEvent.java

```diff
@@ -0,0 +1,146 @@
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests.
+    try (BatchServiceClient batchServiceClient = BatchServiceClient.create()) {
+      TaskSpec task = TaskSpec.newBuilder()
+              // Jobs can be divided into tasks. In this case, we have only one task.
+              .addAllRunnables(buildCustomEvents(displayName1, displayName2, displayName3))
+              .setMaxRetryCount(2)
+              .setMaxRunDuration(Duration.newBuilder().setSeconds(3600).build())
+              .build();
+
+      // Tasks are grouped inside a job using TaskGroups.
+      // Currently, it's possible to have only one task group.
+      TaskGroup taskGroup = TaskGroup.newBuilder()
+          .setTaskCount(3)
+          .setParallelism(3)
+          .setTaskSpec(task)
+          .build();
+
+      Job job =
+          Job.newBuilder()
+              .addTaskGroups(taskGroup)
+              .putLabels("env", "testing")
+              .putLabels("type", "script")
+              // We use Cloud Logging as it's an out of the box available option.
+              .setLogsPolicy(
+                  LogsPolicy.newBuilder().setDestination(Destination.CLOUD_LOGGING))
+              .build();
+
+      CreateJobRequest createJobRequest =
+          CreateJobRequest.newBuilder()
+              // The job's parent is the region in which the job will run.
+              .setParent(String.format("projects/%s/locations/%s", projectId, region))
+              .setJob(job)
+              .setJobId(jobName)
+              .build();
+
+      Job result =
+          batchServiceClient
+              .createJobCallable()
+              .futureCall(createJobRequest)
+              .get(5, TimeUnit.MINUTES);
+
+      System.out.printf("Successfully created the job: %s", result.getName());
+
+      return result;
+    }
+  }
+
+  // Create custom status events
+  private static Iterable<Runnable> buildCustomEvents(String displayName1,
+                                                      String displayName2, String displayName3) {
+    List<Runnable> runnables = new ArrayList<>();
+
+    // Define what will be done as part of the job.
+    runnables.add(Runnable.newBuilder()
+        .setDisplayName(displayName1)
+        .setScript(
+            Script.newBuilder()
+                .setText(
+                    "echo Hello world from script 1 for task ${BATCH_TASK_INDEX}")
+                // You can also run a script from a file. Just remember, that needs to be a
+                // script that's already on the VM that will be running the job.
+                // Using setText() and setPath() is mutually exclusive.
+                // .setPath("/tmp/test.sh")
+                .build())
+        .build());
+
+    runnables.add(Runnable.newBuilder()
+        .setDisplayName(displayName2)
+        .setBarrier(Runnable.Barrier.newBuilder().build())
```

Comment:
Why is it an empty builder here?

Reply:
i copy this sample from this page https://cloud.google.com/batch/docs/create-run-job-custom-status-events

Reply:
maybe it's better to remove this runnable then show empty one

Reply:
No, kindly retain the barrier example to match the docs.
Additionally as per the docs, create the custom log example as well.